Documentation/contributors/design-proposals/installed-api.md

```diff
@@ -0,0 +1,282 @@
+
+type Installed struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Status InstalledStatus `json:"status,omitempty"`
+}
+
+type InstalledStatus struct {
+    // Use specific versions of tracked resources (v1alpha1)
+    Subscription v1alpha1.Subscription `json:"omitempty,subscription"`
+    ClusterServiceVersion v1alpha1.ClusterServiceVersion `json:"omitempty,clusterServiceVersion"`
+}
+
+// <definition of Installed list structure>
+```
+
+The internal type under package `github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/apis/operators`:
+
+```go
+package operators
+
+import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+    "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+)
+
+// Internal type for apiserver use
+
+type Installed struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+
+    Status InstalledStatus
+}
+
+type InstalledStatus struct {
+    // Lock-down specific versions here since they are part of a different API
+    Subscription v1alpha1.Subscription
+    ClusterServiceVersion v1alpha1.ClusterServiceVersion
+}
+
+// <definition of Installed list structure>
+```
+
+Conversions, defaults, open-api definitions, informers, clients, etc. will be autogenerated for these types using the [k8s code-generators](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/README.md#code-generator). The set up should be a fairly simple extension of what currently exists for other `packageserver` types.
+
+The extension api-server will use the internal types, clients, and informers for the API group it owns (`packages.operators.coreos.com`).
+
+### Watching
+
+To support watching, `InstalledList` resources will be assigned a `ResourceVersion` field that is an aggregate of those found in its tracked `<Resource>List`s. The aggregate field will be a comma delimited string of tuples in the format `R0:RV0,R1:RV1,...,Ri:RVi,...,Rn:RVn`, where each `Ri` represents a unique, tracked, API list resource, and `RVi` represents its respective `ResourceVersion`. When a watch is requested, the extension api-server will parse the `ResourceVersion` into its components and open a watch for each API resource. When an event is received from one of these watches, the corresponding `Installed` resources will be `resynthesized` and emitted to the top-level `Installed` watch.
+
+### Deprecating Copied `CSV`s
+
+In the long-term, all logic generating copied `CSV`s will _eventually_ be removed.
+
+Changes removing copied `CSV`s from OLM:
+
+- Remove `CSV` copying during `OperatorGroup` reconciliation
+- Remove copied `CSV` queues, sync functions, and garbage collection
+- Update garbage collection for target namespace RBAC (can no longer add `OwnerReference`s to target `CSV`s)
+  - Add surrogate resource in target namespace to use for `OwnerReference` and garbage collection
```

Comment:
If we still need a surrogate resource, that resource will have need almost exactly the csv copying logic (modulo status updates) so we might not want to throw that out. 

But I would rather use ownerefs to this new API unless that is impossible for some reason